source/manage-apps/configure/config/sync.txt

```diff
@@ -215,88 +219,160 @@ Sync configuration:
-       descriptive name to make it easier to parse the rules, such as 
-       "adminReadWrite" or "employeeReadOnly".
 
-   * - | ``rules.<Coll>.applyWhen``
-       | JSON Expression
-     - An :ref:`expression <expressions>` that evaluates to ``true`` when 
-       the rule should be applied. An example might be 
-       ``{"%%user.custom_data.isGlobalAdmin": true}``. This reads the 
-       value of the ``isGlobalAdmin`` bool from the :ref:`user's custom data 
-       <custom-user-data>`, and applies the rule when the value of this 
-       bool is ``true``.
+     - The name of the role. You might use a descriptive name, such as
+       ``"adminReadWrite"`` or ``"employeeReadOnly"``, to make it easier
+       to understand the rules.
 
-   * - | ``rules.<Coll>.read``
-       | JSON Expression
-     - An :ref:`expression <expressions>` that evaluates to ``true`` when a 
-       user has permission to read objects in the collection. If the 
-       expression evaluates to ``false``, {+service-short+} does not allow 
-       the user to read an object or its properties.
-
-   * - | ``rules.<Coll>.write``
-       | JSON Expression
-     - An :ref:`expression <expressions>` that evaluates to ``true`` when a 
-       user has permission to write objects in the collection. If the 
-       expression evaluates to ``false``, {+service-short+} does not allow 
-       the user to modify an object or its properties. Write permission
-       requires read permission. A user cannot write to a document that
-       they cannot read.
+   * - | ``rules.<Type Name>.[].applyWhen``
+       | Expression
+     
+     - An :ref:`expression <expressions>` that evaluates to ``true``
+       when the role applies to the user. 
+       
+       .. example::
+          
+          The following expression defines a role that applies when a
+          :ref:`user's custom data <custom-user-data>` contains a field
+          named ``isGlobalAdmin`` with the value set to ``true``:
+
+          .. code-block:: json
+             
+             {
+               "%%user.custom_data.isGlobalAdmin": true
+             }
+
+   * - | ``rules.<Type Name>.[].read``
+       | Expression
+     
+     - An :ref:`expression <expressions>` that determines whether or not
+       the user can read data. If the expression evaluates to ``true``,
+       the user has permission to read objects of this type. If
+       ``false``, the user cannot read an object or any of its
+       properties.
+
+   * - | ``rules.<Type Name>.[].write``
+       | Expression
+     
+     - An :ref:`expression <expressions>` that determines whether or not
+       the user can write data. If the expression evaluates to ``true``,
+       the user has permission to write objects of this type. If
+       ``false``, the user cannot write an object or any of its
+       properties.
+       
+       Write permission requires read permission. A user cannot write to
```

Comment:
I don't think Asymmetric Sync affects this. You still need to have permission to view the value of a field in order to modify it even if changes from other clients aren't synced down.

Reply:
I wonder if it's worth calling that out explicitly - maybe on the Asymmetric Sync page? i.e. "Even though Asymmetric Sync is for write-only workloads, the user must still have read permission to write" or something like that.

Maybe it's outside the scope of this PR - just something I was thinking about as I was reading these detailed permission docs.

Reply:
I can try to shed some light here. I think "Write permission requires read permission" is not really accurate, it's more like "Write permission _implies_ read permission". Under the hood, if the read/write permissions are different, then the end result of the client's read permission filter becomes: 
```js
{ "$or": [<readFilter>, <writerFilter>] }
```

So effectively, a client cannot write to an object they cannot read - even if their standalone read permissions indicates otherwise.

w.r.t asymmetric sync, read permissions on asymmetric tables are actually never used. maybe that is worth calling out in the docs somewhere?

> You still need to have permission to view the value of a field in order to modify it even if changes from other clients aren't synced down.

asymmetric tables are "insert only" from sync's perspective. Once the realm write transaction that creates the object is closed, that object becomes effectively immutable, and any attempts to upload modifications to the object will result in an error

Reply:
This info is is definitely helpful! I think we could include it in a new, more in-depth explainer section of the Rules docs.